packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -168,20 +183,39 @@ async def unarchive_dir(
                 for zip_entry in zip_file_handler.infolist()
             ]
 
-            extracted_paths: list[Path] = await tqdm_asyncio.gather(
-                *tasks,
-                desc=f"decompressing {archive_to_extract} -> {destination_folder} [{len(tasks)} file{'s' if len(tasks) > 1 else ''}"
-                f"/{_human_readable_size(archive_to_extract.stat().st_size)}]\n",
-                total=len(tasks),
-                **_TQDM_MULTI_FILES_OPTIONS,
-            )
+            try:
+                extracted_paths: list[Path] = await tqdm_asyncio.gather(
+                    *futures,
+                    desc=f"decompressing {archive_to_extract} -> {destination_folder} [{len(futures)} file{'s' if len(futures) > 1 else ''}"
+                    f"/{_human_readable_size(archive_to_extract.stat().st_size)}]\n",
+                    total=len(futures),
+                    **_TQDM_MULTI_FILES_OPTIONS,
+                )
+
+            except Exception as err:
+                for f in futures:
+                    f.cancel()
+
+                # wait until all tasks are cancelled.
+                await asyncio.gather(*futures, return_exceptions=True)
```

Comment:
minor: this is more a conceptual thing but what about using asyncio.wait [https://docs.python.org/3/library/asyncio-task.html#asyncio.wait](https://docs.python.org/3/library/asyncio-task.html#asyncio.wait) ? is this not more explicit?
```asyncio.wait(*futures, timeout=5 minutes,  return_when=ALL_COMPLETED)```

Reply:
Yes, it is more clear your suggestion. I will change it